### PR TITLE
Project Infrastructure SIG

### PR DESCRIPTION
**Pull request by austinlparker**
This proposal creates a new SIG for project tooling, to aid in the maintenance and development of org-wide tools for OpenTelemetry.

### CODE REVIEW

**Review by trask — APPROVED** [review]
I'd like to participate (happy to be a sponsor as well)

**Comment by svrnm** [comment]
Big +1 for this one and happy to participate.

A question on scope: does "tooling" also include package management (pypi, maven, homebrew, crates, npm, ...) and best practices around them? This is currently owned by the individual SIGs, which should remain that way, but there might be some best practices across them as well we would like to look into (how is code published, by whom, using which mechanism, etc.), which also taps into what SIG security is doing.

**Comment by austinlparker** [comment]
> Big +1 for this one and happy to participate.
> 
> A question on scope: does "tooling" also include package management (pypi, maven, homebrew, crates, npm, ...) and best practices around them? This is currently owned by the individual SIGs, which should remain that way, but there might be some best practices across them as well we would like to look into (how is code published, by whom, using which mechanism, etc.), which also taps into what SIG security is doing.

I think we would want to look at what's currently being used, see if there's opportunities for de-duplication, and perhaps provide best practices around things like tooling for secret management or permissions -- but yeah, I think SIGs should continue to manage their own stuff for the most part? That said, the more we can provide 'out of the box' for SIGs the better imo.

**Comment by trask** [comment]
> I think we would want to look at what's currently being used, see if there's opportunities for de-duplication, and perhaps provide best practices around things like tooling for secret management or permissions -- but yeah, I think SIGs should continue to manage their own stuff for the most part

👍

related: #1293

File: projects/project-tooling.md
Comment (svrnm) [inline projects/project-tooling.md:28]:
```suggestion
- @svrnm
```

Since we have enough sponsors already I am happy to be the first contributor :D

File: projects/project-tooling.md
Comment (svrnm) [inline projects/project-tooling.md:1]: What about "Infrastructure SIG", similar to what [k8s has](https://github.com/kubernetes/community/blob/master/sig-k8s-infra/README.md), or is this too big of a scope what you are thinking of?

File: projects/project-tooling.md
Comment (danielgblanco) [inline projects/project-tooling.md:3]: Is this following the new template?

**Comment by svrnm** [comment]
> but yeah, I think SIGs should continue to manage their own stuff for the most part? That said, the more we can provide 'out of the box' for SIGs the better imo.

Similar to what Docs, Security, End-User are doing this is a "SIG-to-SIG Service" approach, some SIGs might be not interested, but some other things would be happy to be taken off the burden to do package management on their own. For some package managers (like homebrew, linux distros, etc.) a centralized approach might also be helpful.

It might not be the first thing this SIG tackles, but I would like to have it as part of the charter. For my day-to-day job I have looked into some of those already (pypi, homebrew, ansible, crates, ...) and also helped to define some best practices, I am happy to provide this here as well.

File: projects/project-tooling.md
Comment (svrnm) [inline projects/project-tooling.md:17]:
```suggestion
- Develop new tools as needed.
- Project-wide service for release and package management
```

See my comment, I would like to establish this "as a service", so SIGs might be able to delegate the package management and focus on other demands.

**Comment by austinlparker** [comment]
Another note/example of a project that would be really swell for this SIG to tackle - an OTLP/JSON validator in the browser to help people figure out why their OTLP may be malformed.

File: projects/project-tooling.md
Comment (danielgblanco) [reply projects/project-tooling.md:1]: Could that be confused with a potential SIG related to Infrastructure instrumentation or conventions?

**Review by svrnm — APPROVED** [review]

File: projects/project-tooling.md
Comment (jpkrohling) [inline projects/project-tooling.md:6]: Q: if a SIG wants to use a new github integration, would they now come to this SIG, or would they continue coming to the TC? Would this new SIG handle the TC interaction required for this?

**Review by mtwo — APPROVED** [review]

**Review by theletterf — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by jaronoff97** [comment]
I would love to help out here given the work we began on [this issue](https://github.com/open-telemetry/community/issues/1596)! Happy to assist in whatever way i can here.

File: projects/project-tooling.md
Comment (austinlparker) [reply projects/project-tooling.md:1]: How about 'Project Infrastructure'?

File: projects/project-tooling.md
Comment (austinlparker) [reply projects/project-tooling.md]: The eventual goal is that this SIG would provide stuff like this 'as a service' (however we define that) which would automate much of the process. I would like to get humans out of the way as much as possible except for approvals.

**Review by svrnm — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Comment by MSNev** [comment]
As part of semantic conventions there is already a separate Semantic Convention Tooling SIG (7am  Wednesdays), this sounds like the same or at least similar.

**Comment by austinlparker** [comment]
> As part of semantic conventions there is already a separate Semantic Convention Tooling SIG (7am  Wednesdays), this sounds like the same or at least similar.
> 
>  

There's a different scope for the two SIGs.

**Comment by MSNev** [comment]
At the very least it seems like what is being proposed is an overall wrapper for what is occurring in that SIG, as we are developing and discussion the "project" tools that every SIG will use for managing not just semantic conventions but also code generation from the semantic conventions. We are also actively looking for additional contributors / maintainers of the project tools being constructed.

**Comment by MSNev** [comment]
Build Tools: https://github.com/open-telemetry/build-tools/tree/main/semantic-conventions
Weaver: https://github.com/open-telemetry/weaver
Triage Project board: https://github.com/orgs/open-telemetry/projects/84
Weaver Project board: https://github.com/orgs/open-telemetry/projects/74

**Comment by austinlparker** [comment]
> At the very least it seems like what is being proposed is an overall wrapper for what is occurring in that SIG, as we are developing and discussing the "project" tools that every SIG will use for managing not just semantic conventions but also code generation from the semantic conventions. We are also actively looking for additional contributors / maintainers of the project tools being constructed.

Sure, I'm sure these two SIGs will coordinate, but there's still a need for project-wide sustaining engineering on things other than semconv code gen. If you would prefer that this SIG subsumes the semconv one, that's a possibility, but I feel like it'd make more sense for that SIG to work independently so it's not blocking adoption.

**Review by chalin — APPROVED** [review]
LGTM. 

Would it be in scope of this new SIG to consider tooling in support of docs processing (of any form, including for tech docs, specs, etc), and their publication via websites? 

From reading the proposal here, I'm not sure that such docs-related tooling is in scope, but if it is, I'm willing to contribute. If it's not in scope, that's fine, but IMHO, there might be a need for shared resources for docs.

**Comment by austinlparker** [comment]
> LGTM. 
> 
> 
> 
> Would it be in scope of this new SIG to consider tooling in support of docs processing (of any form, including for tech docs, specs, etc), and their publication via websites? 
> 
> 
> 
> From reading the proposal here, I'm not sure that such docs-related tooling is in scope, but if it is, I'm willing to contribute. If it's not in scope, that's fine, but IMHO, there might be a need for shared resources for docs.

I think it'd be in scope, yes.

**Review by danielgblanco — APPROVED** [review]

**Comment by jpkrohling** [comment]
What's the status of this one?

**Comment by austinlparker** [comment]
Just waiting for approvals afaik?

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
> Just waiting for approvals afaik?

Alright, you have one more now :-)

**Comment by svrnm** [comment]
A majority of GC members has approved this SIG proposal :tada:

**Comment by svrnm** [comment]
https://github.com/open-telemetry/community/issues/2186 summarizes the next steps

**Pull request merged**
